src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionPatchFilterElement.java

```diff
@@ -99,13 +110,34 @@ private boolean isLineMatching(AuditEvent event) {
      */
     private boolean isContextStrategyCheck(AuditEvent event) {
         boolean result = false;
-        final String[] checkNames = event.getLocalizedMessage().getSourceName().split("\\.");
-        final String checkShortName = checkNames[checkNames.length - 1];
+        final String checkShortName = getCheckShortName(event);
         if ("RegexpOnFilenameCheck".equals(checkShortName)
                 || "FileLengthCheck".equals(checkShortName)
                 || "NewlineAtEndOfFileCheck".equals(checkShortName)) {
             result = true;
         }
         return result;
     }
+
+    /**
+     * Is matching by never suppress check.
+     *
+     * @param event event
+     * @return true if it is matching
+     */
+    private boolean isNeverSuppressCheck(AuditEvent event) {
+        boolean result = false;
+        final String checkShortName = getCheckShortName(event);
+        if (neverSuppressedChecks != null) {
+            if (neverSuppressedChecks.contains(checkShortName)) {
```

Comment:
Please match by `id` also, please read issue attantively

Reply:
done

Reply:
I meant to use single property that keeps names and ids. From event we take name and I'd and try to match.